src/provider.coffee

```diff
@@ -5,6 +5,25 @@ extensions        = require './extensions'
 
 
 
+paramMatches = (param, duplicate) ->
+  if Array.isArray param
```

Comment:
You're absolutely right - I was rushing so I messed that up.

If we're trying to test if the property contains the exact same contents we probably should do a length check and a more strict comparison. I was thinking over the possible cases and the best course of action would be probably to sort each array and compare the contents. As an example, if the case `[1, 1, 2]` compared to `[1, 2, 1]` we would need to ensure all duplicates were covered.

Does that seem like it would cover the array cases?